src/elm/Generate.elm

```diff
@@ -65,6 +66,130 @@ moduleFromDocs dir prefix module_ =
+            accumulated
+                |> getModuleNamesFromTags first.tags
+                |> getModuleNamesFromUnions rest
+
+        [] ->
+            accumulated
+
+
+{-| -}
+getModuleNamesFromTags : List ( String, List Elm.Type.Type ) -> Set String -> Set String
+getModuleNamesFromTags tags accumulated =
+    case tags of
+        ( _, types ) :: rest ->
+            accumulated
+                |> getModuleNamesFromTypes types
+                |> getModuleNamesFromTags rest
+
+        [] ->
+            accumulated
+
+
+{-| -}
+getModuleNamesFromAliases : List Elm.Docs.Alias -> Set String -> Set String
+getModuleNamesFromAliases aliases accumulated =
+    case aliases of
+        first :: rest ->
+            accumulated
+                |> getModuleNamesFromType first.tipe
+                |> getModuleNamesFromAliases rest
+
+        [] ->
+            accumulated
+
+
+{-| -}
+getModuleNamesFromValues : List Elm.Docs.Value -> Set String -> Set String
+getModuleNamesFromValues values accumulated =
+    case values of
+        first :: rest ->
+            accumulated
+                |> getModuleNamesFromType first.tipe
+                |> getModuleNamesFromValues rest
+
+        [] ->
+            accumulated
+
+
+{-| -}
+getModuleNamesFromTypes : List Elm.Type.Type -> Set String -> Set String
+getModuleNamesFromTypes types accumulated =
+    List.foldl getModuleNamesFromType accumulated types
+
+
+getModuleNamesFromType : Elm.Type.Type -> Set String -> Set String
+getModuleNamesFromType type_ accumulated =
+    case type_ of
+        Elm.Type.Var _ ->
+            accumulated
+
+        Elm.Type.Lambda argument result ->
+            accumulated
+                |> getModuleNamesFromType argument
+                |> getModuleNamesFromType result
+
+        Elm.Type.Tuple nestedTypes ->
+            accumulated
+                |> getModuleNamesFromTypes nestedTypes
+
+        Elm.Type.Type typeName arguments ->
+            accumulated
+                |> addModuleNameFromTypeName typeName
+                |> getModuleNamesFromTypes arguments
+
+        Elm.Type.Record fields _ ->
+            accumulated
+                |> getModuleNamesFromTypes (List.map Tuple.second fields)
+
+
+{-| -}
+addModuleNameFromTypeName : String -> Set String -> Set String
+addModuleNameFromTypeName typeName accumulated =
+    let
+        -- inefficient hackery to get all but the last component of a fully qualified type name
+        moduleName =
+            typeName
+                |> String.split "."
+                |> List.reverse
+                |> List.drop 1
+                |> List.reverse
+                |> String.join "."
+    in
+    if String.isEmpty moduleName || List.member moduleName builtinModuleNames then
+        accumulated
+
+    else
+        Set.insert moduleName accumulated
+
+
+{-| -}
+builtinModuleNames : List String
+builtinModuleNames =
+    [ "Basics", "List", "Maybe", "Result", "String", "Char" ]
```

Comment:
It probably makes sense to just include _all_ the `elm/core` modules here:
```
-   Array
-   Basics
-   Bitwise
-   Char
-   Debug
-   Dict
-   List
-   Maybe
-   Platform
-   Platform.Cmd
-   Platform.Sub
-   Process
-   Result
-   Set
-   String
-   Task
-   Tuple
```
Except probably `Debug`, I guess.

Reply:
I'm not sure...the `builtinModuleNames` list is currently used in `addModuleNameFromTypeName` to figure out which modules _shouldn't_ ever be imported in generated code. You wouldn't want to add things like `Array` and `Dict` to that list, because you _do_ need to import those modules if you need to refer to Elm's `Array` or `Dict` types in a type annotation! But you never need to import `Basics`, `List`, `Maybe`, `Result`, `String` or `Char`.

Reply:
Maybe a better/clearer name would be `alreadyImportedModuleNames` or similar?

Reply:
Ah yeah thats my mistake then, misunderstood what it was for. 

`defaultImportedModules` seems good but maybe that's bikeshedding over a name.

Reply:
If the name caused some confusion then that's definitely a good enough reason to change it! Just switched it to `defaultImportedModuleNames`.